mlir/test/CAPI/CMakeLists.txt

```diff
@@ -5,6 +5,10 @@ set(LLVM_OPTIONAL_SOURCES
   mixr_full.c
 )
 
+set(LLVM_LINK_COMPONENTS
+  X86
+)
+
```

Comment:
Note that this is the blocking factor for removing x86 backend from static lib. Those tests (as executables) should be able to continue to run in static lib scenarios.

This make me to decide to revert the change that get rid of x86 backend. And if we indeed want to push for that, we'll need to make a decision how to deal with future components/tests that would dependent on x86 backend (and in a different PR).

Reply:
... Those tests should be amemned to not need the x86 backend

Reply:
In that, as far as I'm aware, they don't actually use the JIT, so they don't need the x86 backend.

Reply:
I'll cherry-pick dd2b727 and 8309f20 in the next PR to figure out what's causing the linking issue.